packages/client/src/publish/ProxyPublisher.ts

```diff
@@ -0,0 +1,39 @@
+import { StreamDefinition } from '../types'
+import { StreamIDBuilder } from '../StreamIDBuilder'
+import { inject, Lifecycle, scoped } from 'tsyringe'
+import BrubeckNode from '../BrubeckNode'
+import { Context } from '../utils/Context'
+
+@scoped(Lifecycle.ContainerScoped)
+export default class ProxyPublisher {
+    constructor(
+        context: Context,
+        private node: BrubeckNode,
+        @inject(StreamIDBuilder) private streamIdBuilder: StreamIDBuilder,
+    ) {
+    }
+
+    async setPublishProxy(streamDefinition: StreamDefinition, nodeId: string): Promise<void> {
+        const streamPartId = await this.streamIdBuilder.toStreamPartID(streamDefinition)
+        await this.node.openPublishProxyConnectionOnStreamPart(streamPartId, nodeId)
+    }
+
+    async removePublishProxy(streamDefinition: StreamDefinition, nodeId: string): Promise<void> {
+        const streamPartId = await this.streamIdBuilder.toStreamPartID(streamDefinition)
+        await this.node.closePublishProxyConnectionOnStreamPart(streamPartId, nodeId)
+    }
+
+    async setPublishProxies(streamDefinition: StreamDefinition, nodeIds: string[]): Promise<void> {
+        const streamPartId = await this.streamIdBuilder.toStreamPartID(streamDefinition)
+        await Promise.allSettled(
+            nodeIds.map((nodeId) => this.node.openPublishProxyConnectionOnStreamPart(streamPartId, nodeId))
+        )
+    }
+
+    async removePublishProxies(streamDefinition: StreamDefinition, nodeIds: string[]): Promise<void> {
```

Comment:
Would it make sense if the method names would be a pair of `addPublishProxy`+`removePublishProxy`? Or `openPublishProxy`+`closePublishProxy`, as the method names are in `BrubeckNode`.

Reply:
I personally prefer set/remove as the key word here. It feels stricter than add and more appropriate than open/close as the NetworkNode will attempt to maintain a proxy-connection after the initial attempt succeeds